install/kubernetes/cilium/values.yaml

```diff
@@ -1713,51 +1492,41 @@ ipam:
     # refill the bucket up to the burst size capacity.
     # @default -- `4.0`
     externalAPILimitQPS: ~
-
 nodeIPAM:
   # -- Configure Node IPAM
   # ref: https://docs.cilium.io/en/stable/network/node-ipam/
   enabled: false
-
 # -- The api-rate-limit option can be used to overwrite individual settings of the default configuration for rate limiting calls to the Cilium Agent API
 apiRateLimit: ~
-
 # -- Configure the eBPF-based ip-masq-agent
 ipMasqAgent:
   enabled: false
 # the config of nonMasqueradeCIDRs
 # config:
-  # nonMasqueradeCIDRs: []
-  # masqLinkLocal: false
-  # masqLinkLocalIPv6: false
+#   nonMasqueradeCIDRs: []
+#   masqLinkLocal: false
+#   masqLinkLocalIPv6: false
 
 # iptablesLockTimeout defines the iptables "--wait" option when invoked from Cilium.
 # iptablesLockTimeout: "5s"
-
 ipv4:
   # -- Enable IPv4 support.
   enabled: true
-
 ipv6:
   # -- Enable IPv6 support.
   enabled: false
-
 # -- Configure Kubernetes specific configuration
-k8s: {}
+k8s:
   # -- requireIPv4PodCIDR enables waiting for Kubernetes to provide the PodCIDR
   # range via the Kubernetes node resource
-  # requireIPv4PodCIDR: false
-
+  requireIPv4PodCIDR: false
```

Comment:
This is a change in behavior. We are currently using `hasKey .Values.k8s "requireIPv4PodCIDR"` in the config map template to check this field is set and omit the config map key to fall back on the agent default behavior in unset. 

So this will now set it to `false`, thereby adding a previously absent config map key to all new instances of the configmap.

In this case this is not a breaking change (since the agent default is also false). Given the commit description, I assume this is a manual change - but if this is something that `yq` does by default, then we ought to be careful. Replacing absent values with "false" or "null" can break things, since `hasKey` is a common pattern in our Helm templates and the agent might behave differently if certain options are suddenly set to `false`.

Reply:
No, this is one of the options I mentioned I uncommented so that it's preserved after running `yq`. Without, it ends up getting removed, or moved in word way (I forget which). It has the side effect that it's now documented in the README though.

I did verify each of these should retain compatibility, but I didn't adjust the templates (eg: using `hasKey` for this example) to reduce the scope of the change.

Reply:
> By uncommenting It, it has the side effect that it's now documented in the README though, which I thought was an improvement.

Yeah, this is an unfortunate bugs in helm-docs (previous discussion https://github.com/cilium/cilium/issues/21107#issuecomment-1230167839)

Reply:
Ah. Well at least setting a default "fixes" it somewhat.